### ISSUE TITLE
Airbyte Helm Chart: Add registry prefix for all image tag references

### ISSUE THREAD

**Issue by seanglynn-thrive** [airbytehq/airbyte#21123]
## Tell us about the problem you're trying to solve
We are currently running [Airbyte via helm](https://github.com/airbytehq/airbyte/tree/master/charts/airbyte) within our internal AKS cluster. Our infrastructure team currently blocks us from pulling any image references which do not have the explicit registry prefix.
For example: if we want to deploy the popular `busybox` container within our K8s environment:
+ If we reference: `busybox:latest` - 😢  This is blocked as we need the registry prefix.
+ If we reference: `docker.io/busybox:latest` -  😄 This is acceptable as it includes the **'docker.io/'** registry prefix

We succeeded in deploying all of the Airbyte infra components (server, webapp, temporal etc.) by overriding some helm values and K8s definitions. However, we are currently stuck as we are facing this issue while attempting to configure our connectors through the UI.

When we configure a connector (postgres for example), our airbyte service account cannot download the connector container image and we cannot deploy our connector:
```
Caused by: io.temporal.failure.ApplicationFailure: message='Failure executing: POST at: https://10.0.0.1/api/v1/namespaces/airbyte-namespace/pods.
Message: Forbidden! Configured service account doesn't have access. 
Service account may have been revoked. admission webhook "validation.gatekeeper.sh" denied the request: [azurepolicy-...] 
Container image airbyte/source-postgres:1.0.35 for container main has not been allowed
```
As the exception states we cannot pull the image reference: `airbyte/source-postgres:1.0.35`.
If there was some way to prepend the `docker.io/` registry prefix or pass a default registry address, we may be able to continue with our Airbyte deployment.

Other similar errors that occur:
```

[azurepolicy-k8sazure...] Container image alpine/socat:1.7.4.3-r0 for container relay-stderr has not been allowed.
28
[azurepolicy-k8sazure...] Container image alpine/socat:1.7.4.3-r0 for container relay-stdout has not been allowed.
27
[azurepolicy-k8sazure...]  Container image busybox:1.28 for container init has not been allowed.
26
[azurepolicy-k8sazure...] Container image curlimages/curl:7.83.1 for container call-heartbeat-server has not been allowed..', type='io.fabric8.kubernetes.client.KubernetesClientException', nonRetryable=false
```

## Describe the solution you’d like
A clear and concise description of what you want to see happen, or the change you would like to see

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Are you willing to submit a PR?

Remove this with your answer :-)

**Comment by k0t3n** [comment]
Hello @seanglynn-thrive!
Did you solve it?

**Comment by mfsiega-airbyte** [comment]
We're looking into how this could be supported. We'll update within a few weeks with a recommendation.

**Comment by Anna-Katona** [comment]
Hey!
We use our own image registry, and we'd like to have an opportunity to override it too. Is it possible to add env var for this case also?

**Comment by seuf** [comment]
On July 15, docker hub will enforce rate limiting from google IP.
https://cloud.google.com/blog/products/containers-kubernetes/mitigating-the-impact-of-new-docker-hub-pull-request-limits?hl=en

To avoid that we have deployed a docker hub mirror in google artifact registry and we are using it for all our images.

But for airbyte dynamic containers there is no environment variable configuration to override container registry.

**Comment by ana-cafemedia** [comment]
Encountering the same issue in aws.... we'd like to configure ecr pass-through cache for docker image to avoid the rate limitation but there is no good way to accomplish that for Airbyte source/destination containers.
Please, provide a way to override the default docker.io url

**Comment by cgreene-pax** [comment]
We're having a similar issue with a twist, that the only images we can use are the ones mirrored to our internal ECR repo

**Comment by Volodymyr-Kuchinskyi** [comment]
I am having the same issue, since we got rate limits from dockerhub, we'd like to use our own image registry mirror

**Comment by bajnok97** [comment]
I have the same issue deploying in OSS

**Comment by joeinnyc** [comment]
Same issue here, any updates on when this functionality could be released?

**Comment by GeorgiyDemo** [comment]
Same

**Comment by seanglynn-thrive** [comment]
This should be resolved in Airbyte v1.2:
https://github.com/airbytehq/airbyte/releases/tag/v1.2.0

I can close this out if folks can confirm that this is no longer an issue post upgrade?

**Comment by hotsezus** [comment]
While overriding deploy images like `minio/mc` is now a thing and issue #34825 is fixed, overriding connector image registry is still impossible.

The only way to deploy marketplace connector in circumstances described by author is to add custom connector in Airbyte instance settings which references to marketplace connector image stored in private registry (see screenshot)

![Image](https://github.com/user-attachments/assets/ac49b741-a96c-4cf5-ae11-1192906cefa6)

**Comment by JakeCowton** [comment]
> overriding connector image registry is still impossible.

Yeah we're seeing the same. Setting `global.image.registry` doesn't affect the source/destination/connection-related containers.

**Comment by mikemucc** [comment]
Is there an Environment Variable somewhere in the stack to point connectors to a custom registry or custom _base_ registry or could we get one as a quick patch?

**Comment by bajnok97** [comment]
Has anyone been able to resolve this issue? Or any workarounds found? Maybe proxying all the requests to the mirror or something? I am looking for the part of code where the image pull happens but I could not find a solution yet.

**Comment by JakeCowton** [comment]
We got around this by making edits directly to the Airbyte database.
If you prefix `docker_repository` in `actor_definition_version` with your preferred registry, that should get it working (we destroyed & recreated the connections too for good measure).

```sql
UPDATE actor_definition_version
SET docker_repository = '<your_registry>' || docker_repository;
```

I suspect this is easily overwritten, so you might want something like a trigger to handle any changes and ensure the prefix remains.

**Comment by bajnok97** [comment]
and how could you deal with the sidecar and main containers?

**Comment by mikemucc** [comment]
If you're using helm chart v1.2.0 you can set the key `global.image.registry` and it will prefix whatever you put here on all _airbyte_ containers. It just doesn't work for the connector containers.

**Comment by hongbo-miao** [comment]
I can confirm for what @mikemucc mentioned, thanks!
I bumped Helm Chart from v0.350.0 to v1.5.1, now all went through proxy cache.

![Image](https://github.com/user-attachments/assets/90e57c88-7e62-47c3-a694-03c18e041c32)
